### PR TITLE
feat(networkdevice): keep IP from inventory

### PR DESCRIPTION
**Pull request by stonebuzz**
Actually [converter](https://github.com/glpi-project/inventory_format/) merge IP and IPS in same node

From GLPI side, we cannot distinguish between the management IP and other IPS

image

And ```GLPIInentory``` use bad IP to do ```NetInventory```

This PR try to handle IP and IPS

Need https://github.com/glpi-project/inventory_format/pull/89

TU have been prepared for this

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !26397

### CODE REVIEW

File: src/Inventory/Asset/InventoryNetworkPort.php [hunk 1]
```diff
@@ -476,8 +476,11 @@ private function handleUpdates()
                             if ($ip_data == $db_ip_data) {
                                 unset($ips[$ip_key]);
                                 unset($db_addresses[$db_ip_key]);
-                             //result found in db, useless to continue
-                                break 1;
+                                //result found in db, useless to continue
```
Comment (g-bougard) [inline src/Inventory/Asset/InventoryNetworkPort.php:479]: Should we keep the comment if you remove the `break` line ?

Comment (stonebuzz) [reply src/Inventory/Asset/InventoryNetworkPort.php]: done

**Review by cedric-anne — COMMENTED** [review]
This seems ok, but, honestly, I am a bit lost about how we should handle ips/network stuffs.

**Comment by trasher** [comment]
> This seems ok, but, honestly, I am a bit lost about how we should handle ips/network stuffs.

On my side... I'm totally lost, from a while :D

**Review by cedric-anne — CHANGES_REQUESTED** [review]
We discussed about it IRL.

We probably should handle the case where a network equipement would answer to SNMP ping on many addresses. In this case, each address should probably be considered as a management port.

**Comment by g-bougard** [comment]
Indeed the management port in GLPI will only set an ip to be used by netinventory task from glpiinventory. IMHO, if netdiscovery reaches one device from different ip ranges, only one ip there is required. So if one appears during a netdiscovery task and is overwritten by the same task some time later with another ip for another range, we should just assume the last ip is sufficient to inventory the device in netinventory. We don't need to inventory 2 times de same device from 2 different ips.

Maybe "Management port" could be renamed or annotated to reflect it is used in GLPI for SNMP inventory and is not what network administrators can think a network management port is for their all-day job.

**Comment by stonebuzz** [comment]
Closed in favor of : https://github.com/glpi-project/glpi/pull/13935

**Pull request closed without merging**